models/models_ws.go

```diff
@@ -45,6 +45,35 @@ type Signal struct {
 	Code int `json:"code"`
 }
 
+func NewSignal(code int) Signal {
+	return Signal{Code: code}
+}
+
+type Message struct {
+	Code    int         `json:"code"`
+	Payload interface{} `json:"payload,omitempty"`
```

Comment:
As discussed, if you want, you can add `error_payload` here too. Up to you though I don't feel strongly about it.

Reply:
for the payload, you have a function optioin `WithPayload`. Then this can be whatever we want and in cases of error it will be of struct `RespFail`. 
But I see the benefit of another field now. I agree, I'll add a Error field which will be of type `RespFail`